buildpacks/ruby/src/layers/bundle_download_layer.rs

```diff
@@ -1,149 +1,137 @@
+) -> libcnb::Result<(Print<SubBullet<Stdout>>, LayerEnv), RubyBuildpackError> {
+    // TODO switch logging to bullet stream
+    let layer_ref = context.cached_layer(
+        layer_name!("bundler"),
+        CachedLayerDefinition {
             build: true,
             launch: true,
-            cache: true,
-        }
-    }
-
-    fn create(
-        &mut self,
-        _context: &BuildContext<Self::Buildpack>,
-        layer_path: &Path,
-    ) -> Result<LayerResult<Self::Metadata>, RubyBuildpackError> {
-        let bin_dir = layer_path.join("bin");
-        let gem_path = layer_path;
-
-        let mut cmd = Command::new("gem");
-        cmd.args([
-            "install",
-            "bundler",
-            "--version", // Specify exact version to install
-            &self.metadata.version.to_string(),
-        ])
-        .env_clear()
-        .envs(&self.env);
-
-        // Format `gem install --version <version>` without other content for display
-        let short_name = fun_run::display(&mut cmd);
-
-        // Arguments we don't need in the output
-        cmd.args([
-            "--install-dir", // Directory where bundler's contents will live
-            &layer_path.to_string_lossy(),
-            "--bindir", // Directory where `bundle` executable lives
-            &bin_dir.to_string_lossy(),
-            "--force",       // Overwrite if it already exists
-            "--no-document", // Don't install ri or rdoc documentation, which takes extra time
-            "--env-shebang", // Start the `bundle` executable with `#! /usr/bin/env ruby`
-        ]);
-
-        log_step_timed(format!("Running {}", fmt::command(short_name)), || {
-            cmd.named_output().map_err(|error| {
-                fun_run::map_which_problem(error, cmd.mut_cmd(), self.env.get("PATH").cloned())
-            })
-        })
-        .map_err(RubyBuildpackError::GemInstallBundlerCommandError)?;
-
-        LayerResultBuilder::new(self.metadata.clone())
-            .env(
-                LayerEnv::new()
-                    .chainable_insert(Scope::All, ModificationBehavior::Delimiter, "PATH", ":")
-                    .chainable_insert(
-                        Scope::All,
-                        ModificationBehavior::Prepend,
-                        "PATH", // Ensure this path comes before default bundler that ships with ruby, don't rely on the lifecycle
-                        bin_dir,
+            invalid_metadata_action: &|_| {
+                (
+                    InvalidMetadataAction::DeleteLayer,
+                    "invalid metadata".to_string(),
+                )
+            },
+            restored_layer_action: &|old: &Metadata, _| {
+                if let Some(cause) = metadata_diff(old, &metadata) {
+                    (RestoredLayerAction::DeleteLayer, cause)
+                } else {
+                    (
+                        RestoredLayerAction::KeepLayer,
+                        "using cached version".to_string(),
```

Comment:
It might make sense to update this to specifically mention "Bundler" here. This is not necessary as part of this PR (it could also be a separate PR if you like), but all the other layers specify what's being reused (e.g. `metrics agent`, `Ruby version`, `gems`:

```
- Metrics agent
  - Using cached metrics agent
- Ruby version `3.1.3` from `default`
  - Using cached Ruby version
- Bundler version `2.4.17` from `Gemfile.lock`
  - Using cached version
- Bundle install
  - Loading cached gems
```

```suggestion
                        "using cached Bundler version".to_string(),
```